File: my-messenger-android/app/src/main/java/my/messenger/androidclient/api/model/Destination.java
package my.messenger.androidclient.api.model;

/**
 * Created by guilherme on 1/26/2018.
 */

public class Destination {
    private String id;
    private DestinationType type;

    public Destination() {
    }

    public Destination(String id, DestinationType type) {
        this.id = id;
        this.type = type;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public DestinationType getType() {
        return type;
    }

    public void setType(DestinationType type) {
        this.type = type;
    }

    @Override
    public String toString() {
        return "Destination{" +
                "id='" + id + '\'' +
                ", type=" + type +
                '}';
    }
}

File: my-messenger-backend/my.messenger.common/Messaging/Destination.cs
using System;

namespace my.messenger.common.Messaging
{
    public class Destination
    {
        public DestinationType Type { get; set; }
        public String Id { get; set; }

        public Destination(DestinationType type, string id)
        {
            Type = type;
            Id = id;
        }

        public Destination()
        {
        }
    }
}